### DISCUSSION TITLE
Gatsby Head vs React Helmet

### DISCUSSION THREAD

**Discussion by joetm** [gatsbyjs/gatsby#36737] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I followed instructions to replace react-helmet with Gatsby Head, as outlined here:
https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/

It works on the first page load, but when navigating to another Gatsby page, there is no meta info and the page title stays blank.

1) Please do not advertise something as a replacement for a plugin in your docs when it clearly does not fully replace the plugin.

2) How to fix it?

### Reproduction Link

n/a

### Steps to Reproduce

1. Follow docs and replace react-helmet with Gatsby Head.
...

### Expected Result

Metadata presents in all gatsby pages.

### Actual Result

Meta data only injected on first page load, not when navigating to other pages.

### Environment

```shell
System:
    OS: Linux 5.19 Pop!_OS 22.04 LTS
    CPU: (4) x64 Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.2.0 - ~/.nvm/versions/node/v18.2.0/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 8.9.0 - ~/.nvm/versions/node/v18.2.0/bin/npm
  Browsers:
    Chrome: 105.0.5195.125
    Firefox: 105.0
  npmPackages:
    gatsby: ^4.0.0 => 4.24.1
    gatsby-cli: ^4.0.0 => 4.15.1
    gatsby-plugin-image: ^2.15.1 => 2.15.1
    gatsby-plugin-no-sourcemaps: ^3.14.0 => 3.14.0
    gatsby-plugin-nprogress: ^3.14.0 => 3.14.0
    gatsby-plugin-react-helmet: ^4.14.0 => 4.14.0
    gatsby-plugin-sharp: ^4.15.1 => 4.15.1
    gatsby-source-filesystem: ^4.15.0 => 4.15.0
    gatsby-transformer-sharp: ^4.15.1 => 4.15.1
  npmGlobalPackages:
    gatsby-cli: 4.14.0
```

**Comment by LekoArts** [comment]
> Please do not advertise something as a replacement for a plugin in your docs when it clearly does not fully replace the plugin.

It works

> How to fix it?

One can help if a reproduction is provided and the problem described better

**Comment by Abhirup-99** [comment]
It is happening in gatsby.com.

**Reply by lezan** [reply]
They are appended to end of `<head>`
